### PR TITLE
Windows Toolchain

### PR DESCRIPTION
**Pull request by chef-delivery**
This Pull Request was opened by Chef Automate user Ryan Hass

Ready to merge? View this change in Chef Automate and click the Approve button: https://delivery.chef.co/e/chef/#/organizations/engineering-services/projects/omnibus-toolchain/changes/5a87b778-2683-4103-858b-4d38967a658d

### CODE REVIEW

File: config/projects/omnibus-toolchain.rb [hunk 1]
```diff
@@ -50,3 +56,18 @@
   signing_identity "Developer ID Installer: Chef Software, Inc. (EU3VF8YLX2)"
 end
 compress :dmg
+
+msi_upgrade_code = "D607A85C-BDFA-4F08-83ED-2ECB4DCD6BC5"
```
Comment (schisamo) [inline config/projects/omnibus-toolchain.rb]: This code should be unique for this project (can't reuse Chef or ChefDK's). You can generate a new GUID at:
https://www.guidgenerator.com/

File: config/software/omnibus-toolchain.rb [hunk 2]
```diff
@@ -20,14 +20,18 @@
 license :project_license
 
 # gnu utilities
-dependency "patch"
-dependency "bash"
-dependency "make"
-dependency "cacerts"
-dependency "gtar"
+if windows?
+  dependency "msys2core_x64"
```
Comment (schisamo) [inline config/software/omnibus-toolchain.rb]: Don't we need a `msys2core_x86` software def also that is included if we are in 32-bit mode? The `windows_arch_i386?` helper is available for [use in software definitions](https://github.com/chef/omnibus-software/search?utf8=%E2%9C%93&q=windows_arch_i386%3F).

File: omnibus.rb [hunk 3]
```diff
@@ -28,9 +28,16 @@
...
+# ------------------------------
+env_omnibus_windows_arch = (ENV["OMNIBUS_WINDOWS_ARCH"] || "").downcase
+env_omnibus_windows_arch = :x86 unless %w{x86 x64}.include?(env_omnibus_windows_arch)
+
+windows_arch   env_omnibus_windows_arch
+
 # Enable S3 asset caching
 # ------------------------------
-use_s3_caching true
+use_s3_caching false
```
Comment (schisamo) [inline omnibus.rb]: Make sure to switch this back before merge.

File: config/projects/omnibus-toolchain.rb [hunk 4]
```diff
@@ -20,14 +20,20 @@
 license "Apache-2.0"
 license_file "LICENSE"
 
-install_dir "/opt/omnibus-toolchain"
+if windows?
+  install_dir  "#{default_root}/chef/#{name}"
```
Comment (tduffield) [inline config/projects/omnibus-toolchain.rb]: I don't quite follow why we need the `/chef` in here for windows?

File: config/projects/omnibus-toolchain.rb [hunk 5]
```diff
@@ -20,14 +20,20 @@
...
+end
 
 build_version Omnibus::BuildVersion.semver
 build_iteration 1
 
-override :ruby, version: "2.3.1"
-override :git,  version: "2.10.2"
-override :gtar, version: "1.28"
+override :ruby,     version: "2.3.1"
+override :git,      version: "2.8.2"
```
Comment (tduffield) [inline config/projects/omnibus-toolchain.rb]: Is there a reason we down-graded git?

File: config/projects/omnibus-toolchain.rb (hunk 4 above)
Comment (rhass) [inline config/projects/omnibus-toolchain.rb]: Yeah, I am not sure why it was set that way... Probably copy-pasta. If it's at all confusing I don't mind changing it --  In my mind I saw this as the company name and stuff inside being the products.

**Comment by schisamo** [comment]
@rhass You still have the duplicated MSI assets I referenced in:
https://github.com/chef/omnibus-toolchain/pull/43#pullrequestreview-10567622

**Comment by rhass** [comment]
This should be good to go now.

/cc @chef/engineering-services

**Comment by schisamo** [comment]
:shipit:

**Comment by rhass** [comment]
@delivery approve

**Pull request merged**